flowdb/bin/build/0010_schema_public.sql

```diff
@@ -0,0 +1,147 @@
+    of the ingestion of text files.
+
+    Available columns:
+
+      - files:  where information about the ingestion of
+                files is kept.
+
+- d_date:                   General purpose date dimension table.
+- d_time:                   General purpose time dimension table.
+
+----------------------------------------------------
+*/
+CREATE TABLE IF NOT EXISTS public.files(
+    
+    id TEXT PRIMARY KEY,
+    full_path TEXT,
+    directory_path TEXT,
+    specification_path TEXT,
+    target_table TEXT,
+    created TIMESTAMPTZ,
+    processed BOOLEAN,
+    datetime TIMESTAMPTZ,
+    size NUMERIC,
+    status TEXT,
+    record_number NUMERIC,
+    log TEXT
+
+    );
+
+    CREATE TABLE d_date
+    (
+      date_dim_id              INT PRIMARY KEY,
+      date_actual              DATE NOT NULL,
+      epoch                    BIGINT NOT NULL,
+      day_suffix               VARCHAR(4) NOT NULL,
+      day_name                 VARCHAR(9) NOT NULL,
+      day_of_week              INT NOT NULL,
+      day_of_month             INT NOT NULL,
+      day_of_quarter           INT NOT NULL,
+      day_of_year              INT NOT NULL,
+      week_of_month            INT NOT NULL,
+      week_of_year             INT NOT NULL,
+      week_of_year_iso         CHAR(10) NOT NULL,
+      month_actual             INT NOT NULL,
+      month_name               VARCHAR(9) NOT NULL,
+      month_name_abbreviated   CHAR(3) NOT NULL,
+      quarter_actual           INT NOT NULL,
+      quarter_name             VARCHAR(9) NOT NULL,
+      year_actual              INT NOT NULL,
+      first_day_of_week        DATE NOT NULL,
+      last_day_of_week         DATE NOT NULL,
+      first_day_of_month       DATE NOT NULL,
+      last_day_of_month        DATE NOT NULL,
+      first_day_of_quarter     DATE NOT NULL,
+      last_day_of_quarter      DATE NOT NULL,
+      first_day_of_year        DATE NOT NULL,
+      last_day_of_year         DATE NOT NULL,
+      mmyyyy                   CHAR(6) NOT NULL,
+      mmddyyyy                 CHAR(10) NOT NULL,
+      is_std_weekend           BOOLEAN NOT NULL
+    );
+
+
+    INSERT INTO d_date
+    SELECT TO_CHAR(datum,'yyyymmdd')::INT AS date_dim_id,
+           datum AS date_actual,
+           EXTRACT(epoch FROM datum) AS epoch,
+           TO_CHAR(datum,'fmDDth') AS day_suffix,
+           TO_CHAR(datum,'Day') AS day_name,
+           EXTRACT(isodow FROM datum) AS day_of_week,
+           EXTRACT(DAY FROM datum) AS day_of_month,
+           datum - DATE_TRUNC('quarter',datum)::DATE +1 AS day_of_quarter,
+           EXTRACT(doy FROM datum) AS day_of_year,
+           TO_CHAR(datum,'W')::INT AS week_of_month,
+           EXTRACT(week FROM datum) AS week_of_year,
+           TO_CHAR(datum,'YYYY"-W"IW-') || EXTRACT(isodow FROM datum) AS week_of_year_iso,
+           EXTRACT(MONTH FROM datum) AS month_actual,
+           TO_CHAR(datum,'Month') AS month_name,
+           TO_CHAR(datum,'Mon') AS month_name_abbreviated,
```

Comment:
Only need one out of month_name or month_name_abbreviated

Reply:
I guess you don't _need_ either because everything can be extracted from the date. Given the creation of the table is one time, and the storage is negligible though, is the abbreviated name a sufficient convenience?